### PR TITLE
Add germlinecnvcaller

### PR DESCRIPTION
**Pull request by ramprasadn**
## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/createpanelrefs/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, also make a PR on the nf-core/createpanelrefs _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: main.nf [hunk 1]
```diff
@@ -16,9 +16,9 @@ nextflow.enable.dsl = 2
     GENOME PARAMETER VALUES
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
-
 params.fasta = WorkflowMain.getGenomeAttribute(params, 'fasta')
-
+params.fai   = WorkflowMain.getGenomeAttribute(params, 'fai')
+params.dict  = WorkflowMain.getGenomeAttribute(params, 'dict')
```
Comment (maxulysse) [inline main.nf:21]: should we check for `dict`, `fai` only when we use `--tools germlinecnvcaller`?

File: subworkflows/local/germlinecnvcaller_cohort.nf [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+        ch_versions = ch_versions.mix(GATK4_PREPROCESSINTERVALS.out.versions)
+        ch_versions = ch_versions.mix(GATK4_COLLECTREADCOUNTS.out.versions.first())
+        ch_versions = ch_versions.mix(GATK4_ANNOTATEINTERVALS.out.versions)
+        ch_versions = ch_versions.mix(GATK4_FILTERINTERVALS.out.versions)
+        ch_versions = ch_versions.mix(GATK4_INTERVALLISTTOOLS.out.versions)
+        ch_versions = ch_versions.mix(GATK4_DETERMINEGERMLINECONTIGPLOIDY.out.versions)
+        ch_versions = ch_versions.mix(GATK4_GERMLINECNVCALLER.out.versions.first())
+
+    emit:
+        versions = ch_versions
```
Comment (maxulysse) [inline subworkflows/local/germlinecnvcaller_cohort.nf]: We should emit the output from `GATK4_GERMLINECNVCALLER` as well

File: main.nf (hunk 1 above)
Comment (ramprasadn) [reply main.nf:21]: This would just point to null if `genomes` is not defined, right?

**Review by maxulysse — COMMENTED** [review]
Looking good, I'm almost considering pushing your subworkflows to nf-core/modules

File: main.nf (hunk 1 above)
Comment (maxulysse) [reply main.nf:21]: yeah, you're right

File: subworkflows/local/germlinecnvcaller_cohort.nf (hunk 2 above)
Comment (ramprasadn) [reply subworkflows/local/germlinecnvcaller_cohort.nf]: We don't really be using the output anywhere else that's why I didn't add it here.. Don't mind doing it, but I am curious to know your reason. Is it just good practice? Or there are other reason we should do it?

**Comment by ramprasadn** [comment]
I don't think this `cohort` mode analysis will find use in other pipelines, since its something an user will run occasionally to create references.. In that case, its probably easier to use this pipeline to create those files.. But [this workflow](https://github.com/nf-core/raredisease/blob/master/subworkflows/local/variant_calling/call_sv_germlinecnvcaller.nf) calling cnvs in `case` mode could be of use to others.. I can polish that one a bit and add it to modules.

File: subworkflows/local/germlinecnvcaller_cohort.nf (hunk 2 above)
Comment (maxulysse) [reply subworkflows/local/germlinecnvcaller_cohort.nf]: I'd say good practice (plus some other ideas I'm trying to work in)
I need to check I'm actually doing that in the other subworkflows...

Comment (ramprasadn) [reply subworkflows/local/germlinecnvcaller_cohort.nf]: Cool! I will add it 👍🏻

**Review by maxulysse — APPROVED** [review]

**Pull request merged**